src/settings/components/account-settings/delegates.tsx

```diff
@@ -20,23 +20,59 @@ import {
 } from '@zextras/carbonio-design-system';
 import { noop } from 'lodash';
 import { useTranslation } from 'react-i18next';
+import { AccountSettingsPrefs } from '../../../../types';
+import { SettingsSectionProps } from '../utils';
+import { useReset } from '../../hooks/use-reset';
 
 export interface DelegateType extends ItemType {
 	email: string;
 	right: string;
 	label?: string;
 }
 
-type DelegatesProps = {
+export type DelegatesProps = {
+	delegatedSendSaveTarget: AccountSettingsPrefs['zimbraPrefDelegatedSendSaveTarget'];
+	updateDelegatedSendSaveTarget: (
+		updatedValue: AccountSettingsPrefs['zimbraPrefDelegatedSendSaveTarget']
+	) => void;
 	delegates: DelegateType[];
-};
-
-const Delegates = ({ delegates }: DelegatesProps): JSX.Element => {
+} & SettingsSectionProps;
+const Delegates = ({
+	delegatedSendSaveTarget,
+	updateDelegatedSendSaveTarget,
+	resetRef,
+	delegates
+}: DelegatesProps): JSX.Element => {
 	const [t] = useTranslation();
 
 	const [activeDelegate, setActiveDelegate] = useState<string>('0');
 
-	const [activeValue, setActiveValue] = useState('1');
+	const [activeValue, setActiveValue] =
+		useState<AccountSettingsPrefs['zimbraPrefDelegatedSendSaveTarget']>(delegatedSendSaveTarget);
+
+	const radioGroupOnChange = useCallback<(newValue: string) => void>(
+		(newValue) => {
+			function isZimbraPrefDelegatedSendSaveTarget(
+				arg: string
+			): arg is NonNullable<AccountSettingsPrefs['zimbraPrefDelegatedSendSaveTarget']> {
+				return ['owner', 'sender', 'both', 'none'].includes(arg);
```

Comment:
we could extract this array of all possible values in a global const (like DELEGATED_SEND_SAVE_TARGET = ['owner', 'sender', 'both', 'none']).
and then we could use the array to type the pref inside the types, as typeof DELEGATED_SEND_SAVE_TARGET[number].
An finally we could use the same const to check the type assertion here. Doing so we are sure that we are checking for all and only the possible values of the pref, reducing the error of checking inside mistyped strings.

I'm thinking about this because if I add 'something' as value of this array of line 58, nothing tells me it's not a valid value. By introducing a const, we will have the type build from the same array we use across the codebase, and so there is a single point of failure (the constant array)

Reply:
I agree wuth this comment solution